cmake/FindCryptoMiniSat.cmake

```diff
@@ -13,16 +13,78 @@
 # CryptoMiniSat_INCLUDE_DIR - the CryptoMiniSat include directory
 # CryptoMiniSat_LIBRARIES - Libraries needed to use CryptoMiniSat
 
+include(deps-helper)
 
 find_path(CryptoMiniSat_INCLUDE_DIR NAMES cryptominisat5/cryptominisat.h)
 find_library(CryptoMiniSat_LIBRARIES NAMES cryptominisat5)
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(CryptoMiniSat
-  DEFAULT_MSG
-  CryptoMiniSat_INCLUDE_DIR CryptoMiniSat_LIBRARIES)
+set(CryptoMiniSat_FOUND_SYSTEM FALSE)
+if(CryptoMiniSat_INCLUDE_DIR AND CryptoMiniSat_LIBRARIES)
+    set(CryptoMiniSat_FOUND_SYSTEM TRUE)
 
-mark_as_advanced(CryptoMiniSat_INCLUDE_DIR CryptoMiniSat_LIBRARIES)
-if(CryptoMiniSat_LIBRARIES)
-  message(STATUS "Found CryptoMiniSat libs: ${CryptoMiniSat_LIBRARIES}")
+    function(GetVersionPart OUTPUT FILENAME DESC)
+      file(STRINGS ${FILENAME} RES REGEX "^#define CRYPTOMINISAT_VERSION_${DESC}[ \\t]+.*")
+      string(REGEX MATCH "[0-9]+" RES "${RES}")
+      set(${OUTPUT} "${RES}" PARENT_SCOPE)
+    endfunction()
+    GetVersionPart(MAJOR "${CryptoMiniSat_INCLUDE_DIR}/cryptominisat5/cryptominisat.h" "MAJOR")
+    GetVersionPart(MINOR "${CryptoMiniSat_INCLUDE_DIR}/cryptominisat5/cryptominisat.h" "MINOR")
+    GetVersionPart(PATCH "${CryptoMiniSat_INCLUDE_DIR}/cryptominisat5/cryptominisat.h" "PATCH")
+    set(CryptoMiniSat_VERSION "${MAJOR}.${MINOR}.${PATCH}")
+
+    check_system_version("CryptoMiniSat")
+endif()
+
+if(NOT CryptoMiniSat_FOUND_SYSTEM)
+    include(ExternalProject)
+
+    set(CryptoMiniSat_VERSION "5.8.0")
+
+    ExternalProject_Add(
+        CryptoMiniSat-EP
+        PREFIX ${DEPS_PREFIX}
+        URL https://github.com/msoos/cryptominisat/archive/refs/tags/${CryptoMiniSat_VERSION}.tar.gz
+        URL_HASH SHA1=f79dfa1ffc6c9c75b3a33f76d3a89a3df2b3f4c2
+        PATCH_COMMAND patch <SOURCE_DIR>/src/packedmatrix.h ${CMAKE_CURRENT_LIST_DIR}/deps-utils/CryptoMiniSat-patch-ba6f76e3.patch
+        CMAKE_ARGS
+          -DCMAKE_BUILD_TYPE=Release
+          -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+          -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+          -DENABLE_ASSERTIONS=OFF
+          -DENABLE_PYTHON_INTERFACE=OFF
+          -DNOBREAKID=ON
+          -DNOM4RI=ON
+          -DNOSQLITE=ON
+          -DNOZLIB=ON
+          -DONLY_SIMPLE=ON
+          -DSTATICCOMPILE=ON
+    )
+    ExternalProject_Add_Step(
+        CryptoMiniSat-EP
+        cleanup
+        DEPENDEES install
+        COMMAND ${CMAKE_COMMAND} -E remove <BINARY_DIR>/cryptominisat5_simple
+        COMMAND ${CMAKE_COMMAND} -E remove <INSTALL_DIR>/bin/cryptominisat5_simple
+    )
+
+    set(CryptoMiniSat_INCLUDE_DIR "${DEPS_BASE}/include/")
```

Comment:
Again, we should use the installed CMake target for CryptoMiniSat.

Reply:
I'm not sure about this. The naive way would be to call `find_package` at this point in time. This does not work, though, as this searches for the export file that is only generated after CryptoMiniSat has been installed (i.e. at build time).

Reply:
Ah yes. Hmm, well then we have to do it this way.